Web3Swift/Blockchain/ExactBlockChainState.swift

```diff
@@ -0,0 +1,70 @@
+//
+// This source file is part of the Web3Swift.io open source project
+// Copyright 2018 The Web3Swift Authors
+// Licensed under Apache License v2.0
+//
+// PendingBlockChainState.swift
+//
+// Created by Vadim Koleoshkin on 8/06/2018
+//
+
+import Foundation
+
+public final class ExactBlockChainState: BlockChainState {
+    
+    private let number: BytesScalar
+    
+    /**
+     Ctor
+     
+     - parameters:
+        - number: number to be converted to parameter
+     */
+    public init(number: BytesScalar) {
+        self.number = number
+    }
+    
+    /**
+     Ctor
+     
+     - parameters:
+        - hex: `StringScalar` representing bytes of the block number in hex format
+     */
+    public convenience init(hex: StringScalar) {
+        self.init(
+            number: BytesFromHexString(
+                hex: hex
+            )
+        )
+    }
+    
+    /**
+     Ctor
+     
+     - parameters:
+        - hex: `String` representing bytes of the block number in hex format
+     */
+    public convenience init(hex: String) {
+        self.init(
+            hex: SimpleString{
+                hex
+            }
+        )
+    }
+    
+    public func toString() throws -> String {
```

Comment:
Missing documentation here

Reply:
Should we extract the body of this function to separate object to remove duplication of code with `QuantityParameter`.  This could be `StringScalar` called `CompactHexString` and then we add prefix to it with `HexPrefixedString`

Reply:
@rockfridrich let's call it a `TrimmedZeroPrefixString` as in `TrimmedZeroPrefixBytes`. You can implement it using already existing `TrimmedPrefixString`.

Thus `ExactBlockChainState` will look like:
```
let number: StringScalar

init(number: BytesScalar) {
    self.number = PrefixedHexString(
        origin: TrimmedZeroPrefixString(
            origin: UnprefixedHexString(
               bytes: number
            )
        )
    ) 
} 
```

Reply:
@rockfridrich and the exact same code will be used for `QuantityParameter`

Reply:
@rockfridrich can you make that change a part of this pr?